### PR TITLE
Update banner display criteria for App Ratings

### LINKED ISSUE #2884: Display ExperienceFeedbackBanner based on store criteria
## User Story
I want the experience feedback rating banner to appear only under some conditions.

## Acceptance Criteria
- The banner must be displayed if the following conditions are all met:
  - +70% products approved
  - 100% products synced
  - No account error such as suspensions, nor warnings
  - Leave today’s criteria of: 1 conversion

## Implementation Brief
* Update the `ExperienceRatingBanner` component to use the following hooks to get the statistics we need to display the banner.
  * `useMCProductStatistics` to get product statistics.
    * The total number of products can be obtained by adding all values from the `statistics` object returned by the hook.
      * The above will be useful to know the percentage of products approved.
    * For 100% synced, we can check if `not_synced === 0 and active > 0`
  * `useGoogleMCAccount` to get the account details.
    * Use the `isReady` property returned by the hook to determine if there are no account level issues.
  * `useProductsReport` to get the conversion statistics
    * The `data.totals.conversions.value` value can be used to check if there's at least 1 conversion.
    * The type should be paid. `REPORT_SOURCE_PAID`.
* Using the data above, ensure the banner is not rendered if the acceptance criteria are not met or if the banner has already been dismissed, as implemented in #2883.

## Test Coverage
- Add E2E tests to account for the scenarios above to render/hide the banner.
- Appropriate unit tests added if needed.
- Tests should validate logic used to determine banner visibility, including edge cases (e.g., 69% product approval, 0 conversions, etc.).

## Notes
- > Leave today’s criteria of: 1 conversion
  - Shall this condition be removed or kept?
- > 100% products synced
  - Is it based on the 70% of approved products?

### PR DESCRIPTION
**Pull request by ankitrox**
### Changes proposed in this Pull Request:

Closes #2884 .
Closes https://linear.app/a8c/issue/GOOWOO-146/display-experiencefeedbackbanner-based-on-store-criteria

_Replace this with a good description of your changes & reasoning._

### Detailed Test Instructions

1. Go to Marketing > Google for Woocommerce.

2. Open `Network` tab in chrome look for following requests and override the content for those requests in chrom devtools. For more information on how to override the response, refer: https://developer.chrome.com/docs/devtools/overrides

  a. Product Statistics 
  - URL: `wp-json/wc/gla/mc/product-statistics`
  - Content: 
  ```json
    {
      "timestamp": 1678886400,
      "statistics": {
        "active": 1250,
        "expiring": 75,
        "pending": 200,
        "disapproved": 30,
        "not_synced": 0
      },
      "scheduled_sync": 5,
      "loading": false,
      "error": null
    }
  ```

  b. Collection
  - URL: `wp-json/wc/gla/mc/connection`
  - Content:
  ```json
    {
      "id": 5330359695,
      "status": "connected",
      "notification_service_enabled": true,
      "wpcom_rest_api_status": null
    }
  ```

  c. Products
  - URL: `wc/gla/ads/reports/products?*`
  - Content:
  ```json
    {
      "free_listings": null,
      "products": null,
      "intervals": null,
      "totals": {
        "clicks": 0,
        "impressions": 0,
        "conversions": 5
      },
      "next_page": null
    }
  ```

3. Once these overrides are saved, refresh the page and make sure that banner appears.

### Changelog entry

>

### CODE REVIEW

File: js/src/components/experience-rating-banner/index.js [hunk 1]
```diff
@@ -9,9 +9,37 @@ import { external as externalIcon } from '@wordpress/icons';
...
+		statisticsData?.statistics || {}
+	).reduce( ( total, value ) => total + value, 0 );
+
+	const notSyncedProducts = statisticsData?.statistics?.not_synced || 0;
+	const activeProducts = statisticsData?.statistics?.active || 0;
+	const conversions = productsReport?.totals?.conversions?.value || 0;
+
+	const isApproved = ( activeProducts / totalProducts ) * 100 >= 70;
+	const isSynced = notSyncedProducts === 0 && activeProducts > 0;
+	const isReady = isMCAccountReady;
```
Comment (asvinb) [inline js/src/components/experience-rating-banner/index.js]: @ankitrox Any reason we have `isReady` again since it was originally defined [here](https://github.com/woocommerce/google-listings-and-ads/pull/2929/files#diff-bffb3dd8ee52779606946d1198545f2ca76114dfde364a2630d6f0eb6198fb3aR20)? Let's just use `isMCAccountReady`

File: js/src/components/experience-rating-banner/index.js [hunk 2]
```diff
@@ -9,9 +9,37 @@ import { external as externalIcon } from '@wordpress/icons';
...
+
+	const totalProducts = Object.values(
+		statisticsData?.statistics || {}
+	).reduce( ( total, value ) => total + value, 0 );
+
+	const notSyncedProducts = statisticsData?.statistics?.not_synced || 0;
+	const activeProducts = statisticsData?.statistics?.active || 0;
+	const conversions = productsReport?.totals?.conversions?.value || 0;
+
+	const isApproved = ( activeProducts / totalProducts ) * 100 >= 70;
```
Comment (asvinb) [inline js/src/components/experience-rating-banner/index.js]: @ankitrox If there are no products, `0`, there'll be division by zero.

Comment (asvinb) [inline js/src/components/experience-rating-banner/index.js]:
```suggestion
	const isApproved = ( ( activeProducts / totalProducts ) * 100 ) >= 70;
```

File: js/src/components/experience-rating-banner/index.js [hunk 3]
```diff
@@ -9,9 +9,37 @@ import { external as externalIcon } from '@wordpress/icons';
...
 const ExperienceRatingBanner = () => {
+	const { data: statisticsData } = useMCProductStatistics();
+	const { isReady: isMCAccountReady } = useGoogleMCAccount();
+	const { data: productsReport } = useProductsReport( REPORT_SOURCE_PAID );
+
+	const totalProducts = Object.values(
+		statisticsData?.statistics || {}
+	).reduce( ( total, value ) => total + value, 0 );
+
+	const notSyncedProducts = statisticsData?.statistics?.not_synced || 0;
```
Comment (asvinb) [inline js/src/components/experience-rating-banner/index.js]: @ankitrox In many places we have the following:
`statisticsData?.statistics?`

It's better to have a single if condition to check if `statisticsData?.statistics` is defined and have the code that relies on it inside.

File: js/src/components/experience-rating-banner/index.js (hunk 2 above)
Comment (asvinb) [inline js/src/components/experience-rating-banner/index.js]: @ankitrox Let's use a more descriptive name for the constant.
```suggestion
	const hasEnoughActiveProducts = ( activeProducts / totalProducts ) * 100 >= 70;
```

**Review by asvinb — CHANGES_REQUESTED** [review]
@ankitrox I left a few comments, can you kindly check them out please?
Also I'am getting the following notice because my account is not authorized. Any chance we can suppress this message?
image

403 response:
```json
{
    "message": "Unable to retrieve report data: The customer account can't be accessed because it is not yet enabled or has been deactivated.",
    "errors": {
        "CUSTOMER_NOT_ENABLED": "The customer account can't be accessed because it is not yet enabled or has been deactivated.",
        "PERMISSION_DENIED": "The caller does not have permission"
    },
    "report_type": "products",
    "report_query_args": {
        "context": "view",
        "after": {
            "date": "2025-06-01 00:00:00.000000",
            "timezone_type": 1,
            "timezone": "+00:00"
        },
        "before": {
            "date": "2025-06-04 00:00:00.000000",
            "timezone_type": 1,
            "timezone": "+00:00"
        },
        "order": "desc",
        "per_page": 200,
        "fields": [
            "sales",
            "conversions",
            "spend",
            "clicks",
            "impressions"
        ],
        "interval": "day",
        "orderby": "sales"
    }
}
```

Endpoint: `/wp-json/wc/gla/ads/reports/products?after=2025-06-01&before=2025-06-04&fields%5B0%5D=sales&fields%5B1%5D=conversions&fields%5B2%5D=spend&fields%5B3%5D=clicks&fields%5B4%5D=impressions&interval=day&orderby=sales&order=desc&_locale=user`

File: js/src/components/experience-rating-banner/index.js (hunk 2 above)
Comment (ankitrox) [reply js/src/components/experience-rating-banner/index.js]: I think we don't need these brackets.

Comment (asvinb) [reply js/src/components/experience-rating-banner/index.js]: 👍

File: js/src/components/experience-rating-banner/index.js [hunk 4]
```diff
@@ -11,15 +11,46 @@ import { store as preferencesStore } from '@wordpress/preferences';
...
 	const { set } = useDispatch( preferencesStore );
 	const isDismissed = usePreference( BANNER_DISMISSED_KEY );
+	const statistics = statisticsData?.statistics || {};
+
+	const totalProducts = Object.values( statistics ).reduce(
+		( total, value ) => total + value,
+		0
+	);
+
+	const notSyncedProducts = statistics?.not_synced || 0;
```
Comment (asvinb) [inline js/src/components/experience-rating-banner/index.js]: @ankitrox Further to my [comment](https://github.com/woocommerce/google-listings-and-ads/pull/2929#discussion_r2126407522), the idea was to prevent the use of all those optional chaining operator. For e.g we can have a function like this:
```js
const shouldDisplayBanner = () => {
		if ( Object.keys( statistics ).length === 0 ) {
			return false;
		}

		const totalProducts = Object.values( statistics ).reduce(
			( total, value ) => total + value,
			0
		);

		if ( ! totalProducts ) {
			return false;
		}

		const notSyncedProducts = statistics.not_synced;
		const activeProducts = statistics.active;
		const conversions = productsReport?.totals?.conversions?.value || 0;

		const hasEnoughActiveProducts =
			( activeProducts / totalProducts ) * 100 >= 70;
		const hasAllProductsSynced = notSyncedProducts === 0 && activeProducts > 0;
		const hasConversions = conversions > 0;

		return (
			hasEnoughActiveProducts &&
			hasAllProductsSynced &&
			isMCAccountReady &&
			hasConversions
		);
	};
```

File: js/src/components/experience-rating-banner/index.js [hunk 5]
```diff
@@ -11,15 +11,46 @@ import { store as preferencesStore } from '@wordpress/preferences';
...
+	const activeProducts = statistics?.active || 0;
+	const conversions = productsReport?.totals?.conversions?.value || 0;
+
+	const hasEnoughActiveProducts =
+		totalProducts && ( activeProducts / totalProducts ) * 100 >= 70;
+	const isSynced = notSyncedProducts === 0 && activeProducts > 0;
```
Comment (asvinb) [inline js/src/components/experience-rating-banner/index.js]: Let's rename that constant to `hasAllProductsSynced` which is more descriptive.

**Review by asvinb — CHANGES_REQUESTED** [review]

File: js/src/components/experience-rating-banner/index.js (hunk 4 above)
Comment (ankitrox) [reply js/src/components/experience-rating-banner/index.js]: Updated to use this function. Thank you.

File: js/src/components/experience-rating-banner/index.js [hunk 6]
```diff
@@ -11,15 +11,58 @@ import { store as preferencesStore } from '@wordpress/preferences';
...
+
+		return (
+			hasEnoughActiveProducts &&
+			hasAllProductsSynced &&
+			isMCAccountReady &&
+			hasConversions
+		);
+	};
+
+	if ( ! shouldDisplayBanner ) {
```
Comment (asvinb) [inline js/src/components/experience-rating-banner/index.js]: @ankitrox `shouldDisplayBanner` is now a function, and you're testing it as a value here. This will always be false (because a function is truthy), so the banner will never render.

**Review by asvinb — CHANGES_REQUESTED** [review]

File: js/src/components/experience-rating-banner/index.js (hunk 6 above)
Comment (ankitrox) [reply js/src/components/experience-rating-banner/index.js]: Oh my bad! 🤦🏼 
I have corrected it and changed it to function call.
Thanks for spotting it. 🙏🏼

File: tests/e2e/utils/app-ratings-overview.js [hunk 7]
```diff
@@ -49,9 +49,34 @@ export default class AppRatingsOverview extends MockRequests {
 				total: 0,
 				loading: false,
 			} ),
+			this.fulfillProductStatisticsRequest( {
```
Comment (asvinb) [inline tests/e2e/utils/app-ratings-overview.js:52]: @ankitrox I removed those lines, i.e `fulfillProductStatisticsRequest` and `fulfillProductsReport` and the tests still pass. Do we need them?

**Review by asvinb — CHANGES_REQUESTED** [review]
@ankitrox I left a comment about the mock utils. The rest of the code LGTM.

File: tests/e2e/utils/app-ratings-overview.js (hunk 7 above)
Comment (ankitrox) [reply tests/e2e/utils/app-ratings-overview.js:52]: @asvinb Yes, we will need them because if we remove these request mocks, then the tests fails.

Comment (asvinb) [reply tests/e2e/utils/app-ratings-overview.js:52]: I just checked and it's fine to leave them.

**Review by asvinb — APPROVED** [review]

**Comment by ankitguptaindia** [comment]
QA Status: ✅ 
--

Tested this PR as per the AC/PR description and using a Mock request, the Banner displays correctly with mock data, and when modifying/changing each response as described, it hides the banner accordingly. 

> Make MC connection status to incomplete
> In products report, conversions is zero.
> In products statistics, not_synced is greater than 0
> In products statistics, approved percentage is less than 70, use following response for it.



https://github.com/user-attachments/assets/a616b32f-f286-4825-ba0a-3548622b8a51

**Comment by ankitguptaindia** [comment]
@ankitrox I was unable to see the banner when using the rules provided in the PR description, and it works great when importing the rules  [tweak_2025-06-12T11_20_43.983Z.json](https://github.com/user-attachments/files/20709207/tweak_2025-06-12T11_20_43.983Z.json) provided by you in the JSON. 

Could you please check the above rules if any updates are needed? thanks!

**Comment by ankitrox** [comment]
@ankitguptaindia I have updated the test instructions to use chrome devtools method.

**Comment by ankitguptaindia** [comment]
New testing method using Chrome Dev tools is working as expected! 

Screenshot 2025-06-13 at 6 11 57 PM
